pytometry/tools/_humap.py

```diff
@@ -0,0 +1,439 @@
+from scipy.stats import entropy
+from scipy.special import softmax
+# import multiprocessing as mp
+import numpy as np
+from scipy.sparse import csr_matrix, vstack, hstack
+from scanpy.tools._umap import umap as embed_umap
+from anndata import AnnData
+# import matplotlib.pyplot as plt
+from matplotlib.widgets import LassoSelector
+from matplotlib.path import Path
+
+HELPER_VAR = dict()
+
+class _Scale:
+    """
+    Class that contains information about a Scale
+    Properties:
+    self.X
+       datapoints
+    self.T
+       Transition matrix
+    self.I
+       Area of Influence matrix
+    self.W
+       Weight vector
+    self.P
+       Probability distribution
+    self.X_hsne
+       Embedding positions
+    self.lm_ind
+       landmark indices
+    self.parent_scale
+       parent scale (_Scale object)
+
+
+    """
+    def __init__(self, X=None, T=None, I=None, W=None, P=None, X_hsne=None, lm_ind=None, parent_scale=None):
+        self.X = X
+        self.T = T
+        self.I = I
+        self.W = W
+        self.P = P
+        self.X_hsne = X_hsne
+        self.lm_ind = lm_ind
+        self.parent_scale = parent_scale
+
+        self.drilled_scale_list = list()   # TODO remove
+
+
+def humap(
+        adata,
+        beta=100,
+        beta_thresh=1.5,
+        theta=50,
+        num_scales=1,
+        include_root_object=False,
+        verbose=False,
+        copy: bool = False,
+        **kwargs
+):
+    '''
```

Comment:
Please adapt documentation to follow the PEP standard. See error report from Github actions: https://github.com/buettnerlab/pytometry/pull/27/checks#step:6:148